src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdLocationUpdateService.java

```diff
@@ -0,0 +1,102 @@
+                          + "can not find hearing with Id: {}", sscsCaseData.getCcdCaseId(), hmcMessage.getHearingID());
+            return;
+        }
+
+        String updatedVenueId = hmcMessage.getHearingUpdate().getHearingVenueID();
+        Venue venue = findVenue(updatedVenueId);
+        if (isNull(venue)) {
+            log.error("Failed to update location for CCD(Case Id: {}) - "
+                          + "can not find venue with Id {} ", sscsCaseData.getCcdCaseId(), updatedVenueId);
+            return;
+        }
+
+        HearingDetails existingHearingDetails = hearingDetails.get();
+        HearingDetails updatedHearingDetails = HearingDetails.builder()
+            .venue(venue)
+            .hearingDate(existingHearingDetails.getHearingDate())
+            .time(existingHearingDetails.getTime())
+            .adjourned(existingHearingDetails.getAdjourned())
+            .eventDate(existingHearingDetails.getEventDate())
+            .hearingId(existingHearingDetails.getHearingId())
+            .venueId(updatedVenueId)
+            .build();
+
+        // remove hearing with outdated venueId
+        List<Hearing> updatedHearingList = sscsCaseData.getHearings().stream()
+            .filter(hearing -> !hearing.getValue().getHearingId().equals(hmcMessage.getHearingID()))
+            .collect(toList());
+
+        // add hearing with updated venueId
+        updatedHearingList.add(Hearing.builder().value(updatedHearingDetails).build());
+
+        sscsCaseData.setHearings(updatedHearingList);
+        log.info("(Case Id: {}) Venue has been updated from epimsId {} to {} for hearingId {}",
+                 sscsCaseData.getCcdCaseId(), existingHearingDetails.getHearingId(),
+                 updatedVenueId, hmcMessage.getHearingID()
+        );
+    }
+
+    public Venue findVenue(String venueId) {
+
+        Optional<VenueRpcDetails> venue = venueRpcDetailsService.getVenue(venueId);
+
+        if (venue.isEmpty()) {
+            log.error("Can not find venue with Id {}", venueId);
+            return null;
+        }
+
+        VenueDetails venueDetails = venue.get().getVenueDetails();
+        return Venue.builder()
+            .address(Address.builder()
+                         .line1(venueDetails.getVenAddressLine1())
+                         .line2(venueDetails.getVenAddressLine2())
+                         .postcodeAddress(venueDetails.getVenAddressPostcode())
+                         .county(venueDetails.getVenAddressCounty())
+                         .town(venueDetails.getVenAddressTown())
+                         .build())
+            .name(venueDetails.getVenName())
+            .build();
+    }
+
+
+    private Optional<HearingDetails> getHearingDetailsFromHearingList(HmcMessage hmcMessage, SscsCaseData caseData) {
+        return caseData.getHearings().stream()
+            .map(Hearing::getValue)
+            .filter(h -> hmcMessage.getHearingID().equals(h.getHearingId()))
```

Comment:
Minor/Optional: h is not a good variable name either hearing/hearingValue/hearingDetails

Reply:
will do